.github/workflows/workflow_call.yml

```diff
@@ -0,0 +1,61 @@
+name: Integration Tests (Workflow Call)
+on:
+  workflow_dispatch:
+    inputs:
+      dd-trace-go-ref:
+        description: 'The ref to checkout dd-trace-go at'
+        required: false
+        type: string
+        default: main
+  workflow_call:
+    inputs:
+      dd-trace-go-ref:
+        type: string
+        required: true
+        description: 'The ref to checkout dd-trace-go at'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  integration-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [macos, ubuntu, windows]
```

Comment:
I wonder if we _need_ to run on all OS'es there or if we can prune to just `ubuntu` instead... I have no problem testing all three; just the macOS and Windows runners are slow & busy...